packages/playwright-core/src/server/network.ts

```diff
@@ -50,6 +50,8 @@ export function rewriteCookies(cookies: types.SetNetworkCookieParam[]): types.Se
     assert(c.url || (c.domain && c.path), 'Cookie should have a url or a domain/path pair');
     assert(!(c.url && c.domain), 'Cookie should have either url or domain');
     assert(!(c.url && c.path), 'Cookie should have either url or path');
+    assert(!(c.expires && c.expires <= 0 && c.expires !== -1), 'Cookie should have a valid expires, only -1 or a positive number for the unix timestamp in seconds is allowed');
+    assert(!(c.expires && c.expires > 0 && new Date(c.expires * 1000).getFullYear() >= 2100), 'Cookie should have a valid expires, only -1 or a positive number for the unix timestamp in seconds is allowed');
```

Comment:
Let's take max date value from [the spec](https://262.ecma-international.org/5.1/#sec-15.9.1.1) ?